In [1]:
import pandas as pd

In [2]:
shots = 6

In [3]:
df = pd.read_excel("nope_data/gpt_response.xlsx")
df

premise  \
0     The cops had him in their headlights. He ran h...   
1     The cops had him in their headlights. He ran h...   
2     For three nights a comet flared through the de...   
3     For three nights a comet flared through the de...   
4     " He just thinks it's so much bull--. He's not...   
...                                                 ...   
2381  I've worked obviously suspicious gobblers for ...   
2382  And, of course, at that time my dad was dying,...   
2383  And, of course, at that time my dad was dying,...   
2384  I felt grubby just looking at her.  I don't kn...   
2385  I felt grubby just looking at her.  I know wha...   

                                             hypothesis label gpt_response  
0                              He was standing earlier.     E          NaN  
1                              He was standing earlier.     E          NaN  
2     A red smudge couldn't be seen on the moon before.     E          NaN  
3     A red smudge couldn't be seen on the moon before.     E          NaN  
4                                       We are someone.     N          NaN  
...                                                 ...   ...          ...  
2381                       They will react in some way.     E            N  
2382                                Something happened.     E            N  
2383                                Something happened.     E            N  
2384                       Elissa has a favorite hobby.     N            N  
2385                       Elissa has a favorite hobby.     E            N  

[2386 rows x 4 columns]

In [4]:
df_copy = df.loc[lambda df: (df["gpt_response"] == "E") | (df["gpt_response"] == "N")]
# print(df.loc[lambda df: df["gpt_response"] in ["E", "N"]])
df_copy

premise  \
8     "I don't watch them." Puke. " I know what you'...   
9     "I don't watch them." Puke. " I don't know wha...   
10    She was proud of that. The doors were closed. ...   
11    She was proud of that. The doors were closed. ...   
12    There will be no commitment of troops anywhere...   
...                                                 ...   
2381  I've worked obviously suspicious gobblers for ...   
2382  And, of course, at that time my dad was dying,...   
2383  And, of course, at that time my dad was dying,...   
2384  I felt grubby just looking at her.  I don't kn...   
2385  I felt grubby just looking at her.  I know wha...   

                             hypothesis label gpt_response  
8            You are missing something.     E            N  
9            You are missing something.     E            N  
10                  It looked some way.     E            N  
11                  It looked some way.     E            N  
12    Someone is talking about Somalia.     E            E  
...                                 ...   ...          ...  
2381       They will react in some way.     E            N  
2382                Something happened.     E            N  
2383                Something happened.     E            N  
2384       Elissa has a favorite hobby.     N            N  
2385       Elissa has a favorite hobby.     E            N  

[2032 rows x 4 columns]

In [5]:
# number of discarded responses
discarded = len(df) - shots - len(df_copy)
discarded

348

In [9]:
def get_confusion_matrix(shots, df_copy):
    tp = 0
    tn = 0
    fp = 0
    fn = 0

    for i in range(len(df_copy)):
        label = df_copy.iloc[i]['label']
        gpt_response = df_copy.iloc[i]['gpt_response']
        
        if (label == 'E') and (gpt_response == 'E'):
            tp += 1
        elif (label in ['N', 'C']) and (gpt_response == 'N'):
            tn += 1
        elif (label == 'E') and (gpt_response == 'N'):
            fn += 1
        else:
            fp += 1
    
    return tp, tn, fp, fn



In [10]:
def get_metrics(tp, tn, fp, fn):
    accuracy = (tp+tn) / (tp+tn+fp+fn)
    precision = tp / (tp+fp)
    recall = tp / (tp+fn)
    f1_score = 2*precision*recall / (precision+recall)
    return accuracy, precision, recall, f1_score

In [11]:
tp, tn, fp, fn = get_confusion_matrix(shots, df_copy)
print(tp+tn+fp+fn)

2032


In [12]:
accuracy, precision, recall, f1_score = get_metrics(tp, tn, fp, fn)
print(accuracy, precision, recall, f1_score)

0.38484251968503935 0.7749627421758569 0.3211859172328598 0.45414847161572053


I thought it would be best to discard the responses which have not been labelled as "E" or "N", since there doesn't seem to be a way to compare them.(348)
I calculated the accuracy (0.38484251968503935), precision(0.7749627421758569), recall(0.3211859172328598) and f1 score (0.45414847161572053) of the model. 